### PR TITLE
[WIP] Add strawman API for narrative event logging

### PR DESCRIPTION
**Pull request by briehl**
**DO NOT MERGE**

First pass at implementation should go here, too.

### CODE REVIEW

File: NarrativeService.spec [hunk 1]
```diff
@@ -363,4 +363,48 @@ module NarrativeService {
...
+    */
+    typedef structure {
+        string username;
+        string narr_ref;
+        timestamp log_time;
+        string level;
+    } LogContext;
+
+    /*
+        Log event string. Should be one of:
```
Comment (MrCreosote) [inline NarrativeService.spec]: We need enums

File: NarrativeService.spec [hunk 2]
```diff
@@ -363,4 +363,48 @@ module NarrativeService {
...
+        log_event event;
+        LogContext context;
+        UnspecifiedObject metadata;
+    } EventLogParams;
+
+    typedef structure {
+        boolean success;
+    } EventLogResult;
+
+    funcdef log_event(EventLogParams params) returns (EventLogResult) authentication required;
```
Comment (MrCreosote) [inline NarrativeService.spec]: This seems nuts. Any user can log any old crap they want at any old time. The server should be controlling what and when something is logged, not a user.

File: NarrativeService.spec (hunk 1 above)
Comment (briehl) [reply NarrativeService.spec]: YES

File: NarrativeService.spec (hunk 2 above)
Comment (briehl) [reply NarrativeService.spec]: We need some kind of hidden service, then, that does this. @scanon specifically didn't want the Narrative talking directly to whatever we use to store logs, and I agree - adding something that lets the Narrative talk directly to Elastic sounds like a Bad Idea.

But anything we stand up to go in between is going to have the same flaws for the same reason.

**Pull request closed without merging**